include/pybind11/cast.h

```diff
@@ -89,20 +93,134 @@ PYBIND11_NOINLINE inline internals &get_internals() {
         );
         internals_ptr->static_property_type = make_static_property_type();
         internals_ptr->default_metaclass = make_default_metaclass();
+        internals_ptr->instance_base = make_object_base_type(internals_ptr->default_metaclass);
     }
     return *internals_ptr;
 }
 
+/** Class that handles extracting pybind11-registered bases from a python base class.  For a class
+ * that doesn't use python-side multiple inheritance, this will be a single class; if the class has
+ * multiple inheritance it could potentially be more.
+ */
+struct all_type_info : all_type_info_data {
```

Comment:
I'm curious: what is the reason for the separation between all_type_info and all_type_info_data?